packages/interaction/src/interaction.js

```diff
@@ -161,11 +175,29 @@ export const makeParam = (param) => (ix) => {
   ix.params[tempId].tempId = tempId
   ix.params[tempId].key = param.key
   ix.params[tempId].value = param.value
+  ix.params[tempId].asParam = param.asParam
   ix.params[tempId].xform = param.xform
   ix.params[tempId].resolve = param.resolve
   return Ok(ix)
 }
 
+export const makeArgument = (arg) => (ix) => {
+  let tempId = uuid()
+  while (uuidExists(ix, tempId)) {
```

Comment:
remove these whiles (all of them) please, they aren't needed at all. There is more entropy in here than stars in the sky, also its not like we need all that many of them.